### PR TITLE
build pip wheel in wrapper cmake

### PR DESCRIPTION
**Pull request by figroc**
The original `CMakeLists.txt` for python installs the lib and py files directly to python site. They are not versioned and the installation dir is affected by `CMAKE_INSTALL_PREFIX`. While the latter is not necessary bad, the former definitely is and it's bad for redistribution. This PR is partially related to #93.

### CODE REVIEW

**Comment by figroc** [comment]
@googlebot I signed it!

**Comment by cheng-chi** [comment]
@figroc @jmr Is there any plan to merge this branch?

**Comment by calestyo** [comment]
anything new on this?

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-project(s2-geometry)
+project(s2geometry VERSION 0.9.0)
 
 include(CMakeDependentOption)
 include(CheckCXXCompilerFlag)
 include(FeatureSummary)
 include(FindPackageHandleStandardArgs)
 
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```
Comment (jmr) [inline CMakeLists.txt:11]: Is this necessary and why? If it is, add a comment.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,13 +1,14 @@
 cmake_minimum_required(VERSION 3.5)
 
-project(s2-geometry)
+project(s2geometry VERSION 0.9.0)
```
Comment (jmr) [inline CMakeLists.txt:3]: Some of this stuff looks like independent cleanups.

**Review by jmr — CHANGES_REQUESTED** [review]

**Comment by jmr** [comment]
Can you resolve the conflicts?

File: CMakeLists.txt (hunk 1 above)
Comment (figroc) [reply CMakeLists.txt]: It's not necessary. Dropped.

File: CMakeLists.txt (hunk 2 above)
Comment (figroc) [reply CMakeLists.txt:3]: The name and version are used in the `setup.py`.

**Comment by jmr** [comment]
CI is failing: https://travis-ci.org/github/google/s2geometry/jobs/726922409#L873

It looks like `setuptools` needs to be installed first. Can you update `README.md` with new python build instructions and update `.travis.yml` to do `pip install setuptools`?

File: src/python/CMakeLists.txt [hunk 3]
```diff
@@ -16,11 +16,31 @@ swig_link_libraries(pywraps2 ${Python3_LIBRARIES} s2)
...
+                   COMMAND ${CMAKE_COMMAND} -E
+                     copy "${CMAKE_CURRENT_SOURCE_DIR}/__init__.py" ${PYWRAPS2_PKG}
+                   DEPENDS s2 _pywraps2)
+add_custom_command(OUTPUT ${PYWRAPS2_DIST}
+                   COMMAND ${Python3_EXECUTABLE} ${PYWRAPS2_SETUP} bdist_wheel
+                   DEPENDS ${PYWRAPS2_PKG} ${PYWRAPS2_SETUP})
+add_custom_target(bdist_wheel ALL DEPENDS ${PYWRAPS2_DIST})
+
+# Install the package
+install(CODE "execute_process(COMMAND ${Python3_EXECUTABLE} ${PYWRAPS2_SETUP} install
```
Comment (jmr) [inline src/python/CMakeLists.txt:45]: can you keep lines to 80 columns?

Comment (figroc) [reply src/python/CMakeLists.txt]: done

**Comment by jmr** [comment]
I'm assuming this is irrelevant after #313 . If not, reopen.

**Pull request closed without merging**